contracts/stakingv2/modules/Rewards.sol

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2017-2021, bZxDao. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0.
+ */
+
+pragma solidity 0.5.17;
+pragma experimental ABIEncoderV2;
+
+import "./Common.sol";
+
+contract Rewards is Common {
+    function initialize(address target) external onlyOwner {
+        _setTarget(this.addRewards.selector, target);
+        _setTarget(this.getVariableWeights.selector, target);
+        _setTarget(this.totalSupplyStored.selector, target);
+    }
+
+    // note: anyone can contribute rewards to the contract
+    function addRewards(uint256 newBZRX, uint256 newStableCoin) external pausable {
+        if (newBZRX != 0 || newStableCoin != 0) {
+            _addRewards(newBZRX, newStableCoin);
+            if (newBZRX != 0) {
+                IERC20(OOKI).transferFrom(msg.sender, address(this), newBZRX);
+            }
+            if (newStableCoin != 0) {
+                curve3Crv.transferFrom(msg.sender, address(this), newStableCoin);
+                _depositTo3Pool(newStableCoin);
+            }
+        }
+    }
+
+    function _addRewards(uint256 newBZRX, uint256 newStableCoin) internal {
+        (vBZRXWeightStored, iBZRXWeightStored, LPTokenWeightStored) = getVariableWeights();
```

Comment:
should we rename this from BZRX to OOKI for some of the stuff or just keep it as BZRX? It doesn't affect usability, but just for consistency with the rebrand and new token so there isn't any confusion.

Reply:
will do. its just some of the renaming has huge effects and I don't want to break backward compatibility